gadget/device_linux.go

```diff
@@ -48,9 +48,15 @@ func FindDeviceForStructure(ps *LaidOutStructure) (string, error) {
 		byPartlabel := filepath.Join(dirs.GlobalRootDir, "/dev/disk/by-partlabel/", encodeLabel(ps.Name))
 		candidates = append(candidates, byPartlabel)
 	}
-
-	if ps.Label != "" {
-		byFsLabel := filepath.Join(dirs.GlobalRootDir, "/dev/disk/by-label/", encodeLabel(ps.Label))
+	if ps.HasFilesystem() {
+		fsLabel := ps.EffectiveFilesystemLabel()
+		if fsLabel == "" && ps.Name != "" {
```

Comment:
what if both `fsLabel` and `ps.Name` are `""` ? if that's the case then candidates will contain `/dev/disk/by-label/` which will result in the loop below failing with `candidate /dev/disk/by-label/ is not a symlink`, which is a bad user experience IMO, and instead we should just not add anything to candidates in this case in which case if we then have no candidates then the user will see ErrDeviceNotFound

Reply:
Good catch. I'll push a patch